File: src/main/java/lingoncraft/client/CommandProviderWrapper.java
package lingoncraft.client;

import com.mojang.logging.LogUtils;
import dev.langchain4j.agent.tool.Tool;
import dev.langchain4j.memory.chat.MessageWindowChatMemory;
import dev.langchain4j.model.openai.OpenAiChatModel;
import dev.langchain4j.service.AiServices;
import org.slf4j.Logger;

public class CommandProviderWrapper {

    private static final Logger LOGGER = LogUtils.getLogger();

    static class GotoChatCommand {

        private void move_To_Coordinates(int x, int y, int z) {
            LOGGER.info("Moving to coordinates: " + x + " " + y + " " + z);


            baritone.api.BaritoneAPI.getProvider().getPrimaryBaritone().getCommandManager().execute("goto " + x + " " + y + (z > 0 ? " " + z : ""));
        }

        @Tool("Moves the user to provided coordinates")
        public void move_To_Coordinates(int x, int y) {
            move_To_Coordinates(x, y, -1);
        }
    }


}

File: src/main/scala/lingoncraft/client/CommandProviderWrapper.scala
package lingoncraft.client

import baritone.api.BaritoneAPI
import baritone.api.pathing.goals.{GoalBlock, GoalXZ}
import com.mojang.logging.LogUtils
import dev.langchain4j.agent.tool.Tool
<<<<<<< Updated upstream
case class EntityPosition(x: Double, z: Double)
||||||| Stash base
=======
import net.minecraft.util.math.BlockPos
>>>>>>> Stashed changes

object CommandProviderWrapper {
  private val LOGGER = LogUtils.getLogger

  class GotoChatCommand {
    @Tool(Array("""execute baritone commands like `command arg1 arg2 argN` using supplied documentation"""))
    def baritoneRaw(command: String): Boolean = {
      LOGGER.warn("Raw command: " + command)
      BaritoneAPI.getProvider.getPrimaryBaritone.getCommandManager
        .execute(command)
    }

    @Tool(Array("`thisway 1000` then `path` to go in the direction you're facing for a thousand blocks"))
    def baritoneThisWay(n: Int): Boolean = {
      LOGGER.info("thisway")
      BaritoneAPI.getProvider.getPrimaryBaritone.getCommandManager
        .execute(s"thisway ${n}")
    }

    @Tool(
      Array("`Walk to `x z` to start moving the player to a specified position without caring about the vertical position")
    )
    def baritoneWalk(x: Double, z: Double): Unit = {
      LOGGER.info(s"set goal: $x, $z")
      BaritoneAPI.getProvider.getPrimaryBaritone.getCustomGoalProcess
        .setGoalAndPath(new GoalXZ(x.toInt, z.toInt));
    }

    @Tool(Array("Get the position of  the player"))
    def whereAmI(): EntityPosition = {
      LOGGER.info("whereAmI")
      val player =
        BaritoneAPI.getProvider.getPrimaryBaritone.getPlayerContext.player()
      val x = player.lastRenderX
      val z = player.lastRenderZ
      EntityPosition(x, z)
    }


    @Tool(
      Array("`Walk to the surface")
    )
    def baritoneSurface(): Unit = {
      LOGGER.info(s"surface")
      BaritoneAPI.getProvider.getPrimaryBaritone.getCommandManager.execute("surface");
    }

  }

}
